add ontology annotations

MagicMilly opened the pull request:
Adding trait ontologies to contrib including persistent urls and annotations by Laurel Cooper

Commit:
add ontology annotations

Commit:
edited file name

dlebauer reviewed (changes requested):
* [x] is there a reason to use escape (`\`) in the text fields throughout?
* [x] Can you remove the fields that are `null`?
* [x] should the field be named `Comments/Notes.1` or just `Comments/Notes`?

MagicMilly commented:
> * [ ]  is there a reason to use escape (`\`) in the text fields throughout?
> * [ ]  Can you remove the fields that are `null`?
> * [ ]  should the field be named `Comments/Notes.1` or just `Comments/Notes`?

- No reason to use escape (`\`) - this was an error I didn't notice when converting to json
- To clarify, there are no `null` fields that are null for every entry. For example, only `canopy height` has a non-null value for `Method Name 2`. Would you like me to remove this field for every entry _except_ for `canopy height` and also apply this rule to other traits that have null values? See screenshot for example: 
<img width="969" alt="Screen Shot 2020-07-09 at 1 47 53 PM" src="https://user-images.githubusercontent.com/34486154/87089182-dd728780-c1ea-11ea-906d-1dc940000f33.png">

- The first `Comment/Notes` value appears to be related to Trait Ontology, and `Comments/Notes.1` appears to be related to Crop Ontology. I could clarify with Laurel and change the name of these values?

Commit:
removed escape backslash

dlebauer commented on contrib/ontologies.json:
```diff
@@ -0,0 +1,1052 @@
+[
+    {
+        "Variable":"days_to_emergence",
+        "Variable Description":"Planting date: 2017-04-20",
```
description should be something like 'number of days from planting until 1/2 of plants emerged'

Commit:
updated using appropriate structure

Commit:
csv for linking ontologies

MagicMilly commented:
@dlebauer As discussed, I'm adding the csv exported from the ontology_annotations google sheet for a possible way to pull linked ontology data. I renamed the columns and dropped the columns with the few crop ontology annotations since we decided to go with the plant trait ontology. 

Questions:
- How to deal with null fields?
- Should a separate row be added for canopy height due to being the only observationVariable with two different methods? For now there are two columns labeled as methodName.
- Should the observationVariableNames be changed to a different format? Currently using underscores.

dlebauer commented on contrib/ontologies.json:
```diff
@@ -0,0 +1,302 @@
+          {
+            "copyright": "2000-2008 Cold Spring Harbor Laboratory and Cornell University, USA",
+            "description": "Plant Trait Ontology",
+            "documentationURL": "http://purl.obolibrary.org/obo/TO_0006061",
```
* this can be renamed ontology_reference
* if we are only using plant trait ontology, this file should only have one result
* documentationURL should point to a canonical URI for the plant trait ontology, e.g. http://purl.obolibrary.org/obo/to.owl
* ontologyDbId could be something like 'TO' (if this is the common identifier for all terms in the plant trait ontology). This can be any unique value, even an integer, but a name is helpful.

dlebauer commented on contrib/ontology_annotations.csv:
```diff
@@ -0,0 +1,73 @@
+observationVariableName,traitDescription,scaleName,methodName,methodName,ontologyDbId,URL,traitName,comments
```
the goal of this table is to link variable names in the database to ontology terms
* here, ontologyDbId should be renamed ontologyReference; 
* create a new field ontologyDbId that has value "TO". this links to the plant trait ontology record in ontologies.json. this will be the same in every row
* Add observationVariableDbId to this table
* later, when constructing the observationvariable response, use this table to generate the ontologyreference

dlebauer committed:
Update ontologies.json

just document a single record

dlebauer committed:
Rename contrib/ontologies.json to contrib/ontologies/ontologies.json

dlebauer committed:
Rename contrib/ontology_annotations.csv to contrib/ontologies/variables_annotations.csv

MagicMilly committed:
Rename ontologies.json to ontology_reference.json

Commit:
added variable ids, to column, and changed term column name

dlebauer commented:
remaining TODO:

* [ ] populate the observationVariableId field where appropriate. Why are some missing? Should SPAD_659 and absorbance_650 be the same?
* [ ] remove methodName field

dlebauer committed:
Update variables_annotations.csv

MagicMilly committed:
Delete variables_annotations.csv

MagicMilly committed:
Add files via upload

MagicMilly commented:
closing this PR and creating new branch due to merge conflicts and other errors
